### DISCUSSION TITLE
Tasmota script: how put a counter value to variable ?

### DISCUSSION THREAD

**Discussion by Korthy2** [arendst/Tasmota#15979] [Q&A] [answered]
GPIO5 have D0 Signal.
Counter1 count

I will the Counter put in a Variable.
But the Variable is erver 0

My Script:

">D"
"a=0"
">B"
"a=Counter1"
">W"
"<center>a = %a%<br>"

Script End

UI Ausgabe:
Counter1 = 4554
a = 0

An idea?

**Comment by sfromis** [comment]
Not using scripting, but based on examples, I'd expect this to capture a counter value at TelePeriod, not at boot time:
```
>T
a=Counter#C1
```

**Comment by pkkrusty** [comment, marked as answer]
a=pc[1]

From scripting page: pc[x] = pulse counter value (x = 1..4)

**Comment by Korthy2** [comment]
thank you very much, that works!

**Comment by cristidobrescu** [comment]
Hi @Korthy2  would you be kind and post the correc script code?  As I didn`t get it what @pkkrusty said about it to modify

**Comment by Korthy2** [comment]
Here my script, after >B  (currentd1=pc[1]

)

 

>D

h1=0

g:currentd1=0

 

>B

currentd1=pc[1]

 

; hours

if (hours>h1 or hours==0) and mins==0

then

h1=hours

currenth1=total/1000

endif

currenth=total/1000-currenth1

 

 

Best Regards

 

Andreas Korth

 

Konrad-Hofmann-Str. 5

76872 Hergersweiler

 

Von: cristid ***@***.***> 
Gesendet: Sonntag, 12. Mai 2024 10:35
An: arendst/Tasmota ***@***.***>
Cc: Korthy2 ***@***.***>; Mention ***@***.***>
Betreff: Re: [arendst/Tasmota] Tasmota script: how put a counter value to variable ? (Discussion #15979)

 

Hi @Korthy2 <https://github.com/Korthy2>  would you be kind and post the correc script code? As I didn`t get it what @pkkrusty <https://github.com/pkkrusty>  said about it to modify

**Comment by cristidobrescu** [comment]
You mean, this is the whole code? : 
 `>D

h1=0

g:currentd1=0
>B
(currentd1=pc[1])



; hours

if (hours>h1 or hours==0) and mins==0

then

h1=hours

currenth1=total/1000

endif

currenth=total/1000-currenth1`




....cause it seem it doesn`t show anything
![Screenshot 2024-05-14 at 20 21 34](https://github.com/arendst/Tasmota/assets/99839121/572f0608-a584-4c46-ab73-016b389a8533)

**Comment by Korthy2** [comment]
This is the Code for tasmota Skript.First define a variable :g:currentd1=0Now you can put the counter PC[1] Content put in the variable:currentd1=pc[1]BRAndreas
-------- Ursprüngliche Nachricht --------Von: cristid ***@***.***> Datum: 14.05.24  19:18  (GMT+01:00) An: arendst/Tasmota ***@***.***> Cc: Korthy2 ***@***.***>, Mention ***@***.***> Betreff: Re: [arendst/Tasmota] Tasmota script: how put a counter value to variable ? (Discussion #15979) 
You mean, this is the whole code? :

D

h1=0
g:currentd1=0

B
(currentd1=pc[1])

; hours
if (hours>h1 or hours==0) and mins==0
then
h1=hours
currenth1=total/1000
endif
currenth=total/1000-currenth1
....cause it seem it doesn`t show anything
